src/dune_rules/gen_rules.ml

```diff
@@ -374,6 +374,21 @@ let has_rules subdirs f =
        ; rules
        })
 
+let rec is_under_melange_emit dir =
+  match Path.Build.parent dir with
+  | None -> Memo.return false
+  | Some dir -> (
+    Only_packages.stanzas_in_dir dir >>= function
+    | None -> is_under_melange_emit dir
+    | Some { Dune_file.stanzas; dir = _; project = _ } -> (
+      match
+        List.exists stanzas ~f:(function
+          | Melange_emit _ -> true
```

Comment:
This check isn't quite right. We should be checking if we're under `$directory-of-melanage-stanzar/$target-field-of-melange-stanza`

Reply:
Good point. I think the current check is only valid after we remove the `target` field in https://github.com/ocaml/dune/pull/6541.

Pushed [5216189](https://github.com/ocaml/dune/pull/6579/commits/5216189d8dca993c5101de3116fb03550480e231)